pkg/obfuscate/appsec.go

```diff
@@ -0,0 +1,538 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package obfuscate
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+// ObfuscateAppSec obfuscates the given appsec tag value in order to remove
+// sensitive values from the appsec security events.
+func (o *Obfuscator) ObfuscateAppSec(val string) string {
+	keyRE := o.opts.AppSec.ParameterKeyRegexp
+	valueRE := o.opts.AppSec.ParameterValueRegexp
+	if keyRE == nil && valueRE == nil {
+		return val
+	}
+	ob := appsecEventsObfuscator{
+		keyRE:   keyRE,
+		valueRE: valueRE,
+	}
```

Comment:
> Have you considered adding a field to `Obfuscator` with this obfuscator? Why make a new one on every call? Isn't it a waste? It's not like it changes between calls. Have you considered this?

Done in https://github.com/DataDog/datadog-agent/pull/11324/commits/11c457b0780a7cc343f26f42329eb8bda61771da

> Also, the `compileRegexp` function in `cmd/trace-agent` scares me a bit because it returns `nil`. I find that a bit dangerous.
> 
> I'd suggest passing simply strings as part of `opts.AppSec.ParamterKey*` and compiling them here using a similar function that also returns the error. You can feel free to ignore the error if you wish.

Given we are doing the obfuscator this way for performance reasons, I didn't want to compile the regular expressions on the "hot path" and this is why they are passed compiled from its client package. `nil` is meant to be useful here as the "disabled regexp" case and is part of the tests.